src/main/java/io/temporal/internal/statemachines/LocalActivityStateMachine.java

```diff
@@ -0,0 +1,304 @@
+      Functions.Proc1<CancellableCommand> commandSink,
+      Functions.Proc1<StateMachine> stateMachineSink) {
+    super(STATE_MACHINE_DEFINITION, commandSink, stateMachineSink);
+    this.replaying = replaying;
+    this.setCurrentTimeCallback = setCurrentTimeCallback;
+    this.localActivityParameters = localActivityParameters;
+    this.localActivityRequestSink = localActivityRequestSink;
+    this.callback = callback;
+    explicitEvent(ExplicitEvent.CHECK_EXECUTION_STATE);
+    explicitEvent(ExplicitEvent.SCHEDULE);
+  }
+
+  enum ExplicitEvent {
+    CHECK_EXECUTION_STATE,
+    SCHEDULE,
+    GET_REQUEST,
+    HANDLE_RESPONSE,
+    NON_REPLAY_WORKFLOW_TASK_STARTED
+  }
+
+  enum State {
+    CREATED,
+    REPLAYING,
+    EXECUTING,
+    REQUEST_PREPARED,
+    REQUEST_SENT,
+    RESULT_NOTIFIED,
+    MARKER_COMMAND_CREATED,
+    MARKER_COMMAND_RECORDED,
+    WAITING_MARKER_EVENT,
+    RESULT_NOTIFIED_REPLAYING
+  }
+
+  public static final StateMachineDefinition<State, ExplicitEvent, LocalActivityStateMachine>
+      STATE_MACHINE_DEFINITION =
+          StateMachineDefinition.<State, ExplicitEvent, LocalActivityStateMachine>newInstance(
+                  "LocalActivity", State.CREATED, State.MARKER_COMMAND_RECORDED)
+              .add(
+                  State.CREATED,
+                  ExplicitEvent.CHECK_EXECUTION_STATE,
+                  new State[] {State.REPLAYING, State.EXECUTING},
+                  LocalActivityStateMachine::getExecutionState)
+              .add(
+                  State.EXECUTING,
+                  ExplicitEvent.SCHEDULE,
+                  State.REQUEST_PREPARED,
+                  LocalActivityStateMachine::sendRequest)
+              .add(State.REQUEST_PREPARED, ExplicitEvent.GET_REQUEST, State.REQUEST_SENT)
+              .add(
+                  State.REQUEST_SENT,
+                  ExplicitEvent.NON_REPLAY_WORKFLOW_TASK_STARTED,
+                  State.REQUEST_SENT)
+              .add(
+                  State.REQUEST_SENT,
+                  ExplicitEvent.HANDLE_RESPONSE,
+                  State.MARKER_COMMAND_CREATED,
+                  LocalActivityStateMachine::createMarker)
+              .add(
+                  State.MARKER_COMMAND_CREATED,
+                  CommandType.COMMAND_TYPE_RECORD_MARKER,
+                  State.RESULT_NOTIFIED,
+                  LocalActivityStateMachine::notifyResultFromResponse)
+              .add(
+                  State.RESULT_NOTIFIED,
+                  EventType.EVENT_TYPE_MARKER_RECORDED,
+                  State.MARKER_COMMAND_RECORDED)
+              .add(State.REPLAYING, ExplicitEvent.SCHEDULE, State.WAITING_MARKER_EVENT)
+              .add(
+                  State.WAITING_MARKER_EVENT,
+                  EventType.EVENT_TYPE_MARKER_RECORDED,
+                  State.MARKER_COMMAND_RECORDED,
+                  LocalActivityStateMachine::notifyResultFromEvent)
+              .add(
+                  // This is to cover the following edge case:
```

Comment:
had a hard time following this. when calling getExecutionState wouldn't it still be as if the "first time" it ran? OR does the force workflowtask is started create a persistent checkpoint

Reply:
Forced workflow task is a workflow task that is recorded in the history the same way as a normal workflow task.